libs/js-sdk/src/update-strategies/createManualUpdateStrategy.ts

```diff
@@ -10,25 +10,34 @@ export function createManualUpdateStrategy(
     httpEndpoint: string,
 ): EffectiveConfigUpdateStrategy {
     let etag: undefined | string
-    let fetchUpdatesSingle: undefined | (() => Promise<void>)
+    let fetchUpdatesSingle:
+        | undefined
+        | (() => Promise<{ error: Error | undefined }>)
 
     return {
         async connect(stateStore) {
             // Force update
             etag = undefined
             // Ensure that we don't trigger another request while one is in flight
-            fetchUpdatesSingle = createEnsureSingle(async () => {
-                etag = await fetchFeaturesConfigurationViaHttp(
+            fetchUpdatesSingle = createEnsureSingleWithBackoff(async () => {
+                const response = await fetchFeaturesConfigurationViaHttp(
                     httpEndpoint,
                     stateStore.audiences,
                     environmentApiKey,
                     stateStore,
                     etag,
                     () => stateStore.audiences,
                 )
+                etag = response.etag
+                return response
             })
 
-            return fetchUpdatesSingle()
+            return fetchUpdatesSingle().then((response) => {
+                if (response.error) {
```

Comment:
Does this change the return type? Or was it just silently failing before?

Reply:
The only thing that has changed is the fetchFeaturesConfigurationViaHttp is no longer throwing an error for TooManyRequests, it is still throwing error for everything which would be caught later in the retry